### PR TITLE
[ESIMD] Allow constructing simd_view from simd

### PR DESCRIPTION
**Pull request by DenisBakhvalov**
The patch allows the following code to work:
```
  simd<int, 16> s = 0;
  simd_view v = s;
```

I added default template parameter for simd_view_impl class
and put it as a last template argument.

### CODE REVIEW

**Comment by DenisBakhvalov** [comment]
The corresponding E2E test is here: https://github.com/intel/llvm-test-suite/pull/454

File: sycl/test/esimd/simd_view.cpp
Comment (kbobrovs) [inline sycl/test/esimd/simd_view.cpp:75]: please add a test case like simd_view v(s);
Can this be disabled and diagnosed - simd_view v(simd<int,8>()); ?

Comment (DenisBakhvalov) [reply sycl/test/esimd/simd_view.cpp]: Added both test cases.

**Review by kychendev — APPROVED** [review]

File: sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp:48]: Do length 1 specializations support this form of constructor? I believe they should as well.

File: sycl/test/esimd/simd_view.cpp
Comment (kbobrovs) [inline sycl/test/esimd/simd_view.cpp:74]: please a test case for length 1

File: sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp
Comment (DenisBakhvalov) [reply sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp:48]: Currently no. If I add this constructor, then we still cannot write:
```
simd<int, 1> s = 0;
simd_view v(s); // error
simd_view<simd<int, 1>, int, 1, 1> v(s); // OK
```
Since that will require having default arguments in the length 1 specialization classes. And default template argument in a class template partial specialization are not allowed([link](https://stackoverflow.com/questions/18700558/default-template-parameter-partial-specialization)).

I haven't yet found a way how we could allow this for length 1 specializations. We may still need it in the future, but for now, the w/a is simple: `auto v = s[0]`.

**Comment by kbobrovs** [comment]
@DenisBakhvalov, could you please hold this until this big one is merged https://github.com/intel/llvm/pull/4230 ?

**Comment by kbobrovs** [comment]
@DenisBakhvalov, #4230 merged, please resolve conflicts then

**Comment by bader** [comment]
@kbobrovs, @kychendev, @sndmitriev, do you need someone else to resolve merge conflicts?

**Comment by bader** [comment]
> @kbobrovs, @kychendev, @sndmitriev, do you need someone else to resolve merge conflicts?

Ping.

File: sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp:76]: Please convert this to a documentation comment, and mention that this does not work for VL=1

Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp:76]: Thanks. Is VL != 1 limitation still in effect? If so the best way would be to use SFINAE to disable this constructor in this case.

Comment (v-klochkov) [reply sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp:76]: I added constructor to scalar specialization. So, VL=1 case works. See this commit: 33eede79f2ca2d92d21dca4b85ca2227bebeb47d

File: sycl/test/esimd/simd_view.cpp
Comment (v-klochkov) [reply sycl/test/esimd/simd_view.cpp]: Added here: 33eede79f2ca2d92d21dca4b85ca2227bebeb47d

**Review by kbobrovs — APPROVED** [review]

**Comment by kbobrovs** [comment]
@v-klochkov, please add a E2E testcase and put a link here. Forgot to mention this during the review.

**Pull request merged**
